azure/init-subscription/enable_feature.sh

```diff
@@ -19,16 +19,16 @@ az feature register --namespace "Microsoft.ContainerService" --name "EnablePodId
 
 while true; do
     result_encryption=$(az feature show --namespace Microsoft.Compute --name EncryptionAtHost | jq -r .properties.state)
-    result_pod_identity=$(az feature show --namespace Microsoft.Compute --name EncryptionAtHost | jq -r .properties.state)
-    
+    result_pod_identity=$(az feature show --namespace Microsoft.ContainerService --name EnablePodIdentityPreview | jq -r .properties.state)
+
     if [ "$result_encryption" == "Registered" ] && [ "$result_pod_identity" == "Registered" ]; then
         echo "[INFO] Feature enabled"
         break
     else
         echo "[INFO] Awaiting feature registration"
-        sleep 20
+        sleep 60
     fi
 done
 
-az provider register -n Microsoft.Compute
-az provider register -n Microsoft.ContainerService
\ No newline at end of file
+az provider register -n Microsoft.Compute --wait
```

Comment:
why do you want to wait result? without waiting you can do something else during the activation

Reply:
I added the wait to ensure that the command completes with the full activation of the provider. If I remember correctly, without the wait flag, I encountered issues with the immediate creation of a VM/AKS, but your observation is correct.